include/GoogleSync/GoogleSync.php

```diff
@@ -0,0 +1,1230 @@
+     */
+    public function updateGoogleCalendarEvent($event_local, $event_remote)
+    {
+
+        if ( (!isset($event_local) || empty($event_local)) || (!isset($event_remote) || empty($event_remote)) )
+        {
+            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'ERROR:Missing Variables');
+            return false;
+        }
+
+        $event_remote->setSummary($event_local->name);
+        $event_remote->setDescription($event_local->description);
+        $event_remote->setLocation($event_local->location);
+
+        $startDateTime = new Google_Service_Calendar_EventDateTime;
+        $startDateTime->setDateTime(date(DATE_ATOM, strtotime($event_local->date_start)));
+        $startDateTime->setTimeZone($this->timezone);
+        $event_remote->setStart($startDateTime);
+
+        $endDateTime = new Google_Service_Calendar_EventDateTime;
+        $endDateTime->setDateTime(date(DATE_ATOM, strtotime($event_local->date_end)));
+        $endDateTime->setTimeZone($this->timezone);
+        $event_remote->setEnd($endDateTime);
+
+        // We pull the existing extendedProperties, and change our values
+        // That way we don't mess with anything else that's using other values.
+        $extendedProperties = $event_remote->getExtendedProperties();
+
+        if (!empty($extendedProperties))
+        {
+            $private = $extendedProperties->getPrivate();
+        }
+        else {
+            $extendedProperties = new Google_Service_Calendar_EventExtendedProperties;
+            $private = array();
+        }
+
+        $private['suitecrm_id'] = $event_local->id;
+        $private['suitecrm_type'] = $event_local->module_name;
+
+        $extendedProperties->setPrivate($private);
+        $event_remote->setExtendedProperties($extendedProperties);
+
+        // Copy over popup reminders
+        $event_local_id = $event_local->id;
+        $reminders_local = BeanFactory::getBean('Reminders')->get_full_list("", "reminders.related_event_module = 'Meetings' AND reminders.related_event_module_id = '$event_local_id' AND popup = '1'");
+
+        if ($reminders_local)
+        {
+            $reminders_remote = new Google_Service_Calendar_EventReminders;
+            $reminders_remote->setUseDefault(false);
+            $reminders_array = array();
+            foreach ($reminders_local as $reminder_local)
+            {
+                $reminder_remote = new Google_Service_Calendar_EventReminder;
+                $reminder_remote->setMethod('popup');
+                $reminder_remote->setMinutes($reminder_local->timer_popup / 60);
+                $reminders_array[] = $reminder_remote;
+            }
+            $reminders_remote->setOverrides($reminders_array);
+            $event_remote->setReminders($reminders_remote);
+        }
+        return $event_remote;
+    }
+
+    /**
+     * Create New Google Event object for SuiteCRM Meeting
+     *
+     * @param $event_local SuiteCRM Meeting Object
```

Comment:
Missing class type. I also noticed that event_local is defined as an object in some code blocks. Is the class type a Bean? Please update PHP Doc.

Reply:
Working on this.